src/purify.spec.ts

```diff
@@ -1,16 +1,104 @@
+      var ChangeDetectionStrategy = /*@__PURE__*/(function() {
+      var ChangeDetectionStrategy = {};
+      ChangeDetectionStrategy.OnPush = 0;
+      ChangeDetectionStrategy.Default = 1;
+      ChangeDetectionStrategy[ChangeDetectionStrategy.OnPush] = "OnPush";
+      ChangeDetectionStrategy[ChangeDetectionStrategy.Default] = "Default";;
+      return ChangeDetectionStrategy;})();
     `;
 
-    expect(oneLine`${purify(input)}`).toEqual(output);
+    expect(oneLine`${purify(input)}`).toEqual(oneLine`${output}`);
+  });
+
+  it('wraps ts 2.3 enums in IIFE', () => {
+    const input = stripIndent`
+      var ChangeDetectionStrategy;
+      (function (ChangeDetectionStrategy) {
+        ChangeDetectionStrategy[ChangeDetectionStrategy["OnPush"] = 0] = "OnPush";
+        ChangeDetectionStrategy[ChangeDetectionStrategy["Default"] = 1] = "Default";
+      })(ChangeDetectionStrategy || (ChangeDetectionStrategy = {}));
+    `;
+    const output = stripIndent`
+      var ChangeDetectionStrategy = /*@__PURE__*/(function() {
+          var ChangeDetectionStrategy = {};
+          ChangeDetectionStrategy[ChangeDetectionStrategy["OnPush"] = 0] = "OnPush";
+          ChangeDetectionStrategy[ChangeDetectionStrategy["Default"] = 1] = "Default";
+          return ChangeDetectionStrategy;
+      })();
+    `;
+
+    expect(oneLine`${purify(input)}`).toEqual(oneLine`${output}`);
+  });
+
+  it('prefixes safe imports with /*@__PURE__*/', () => {
+    const input = stripIndent`
+      /* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__angular_http__ = __webpack_require__(72);
+      /** PURE_IMPORTS_START _angular_http PURE_IMPORTS_END */
+    `;
+    const output = stripIndent`
+      /* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__angular_http__ = /*@__PURE__*/__webpack_require__(72);
+      /** PURE_IMPORTS_START _angular_http PURE_IMPORTS_END */
+    `;
+
+    expect(oneLine`${purify(input)}`).toEqual(oneLine`${output}`);
+  });
+
+  it('prefixes safe default imports with /*@__PURE__*/', () => {
+    // tslint:disable:max-line-length
+    const input = stripIndent`
+      /* harmony import */ var __WEBPACK_IMPORTED_MODULE_0_zone_js___default = __webpack_require__.n(__WEBPACK_IMPORTED_MODULE_0_zone_js__);
+      /** PURE_IMPORTS_START zone_js PURE_IMPORTS_END */
+    `;
+    const output = stripIndent`
+      /* harmony import */ var __WEBPACK_IMPORTED_MODULE_0_zone_js___default = /*@__PURE__*/__webpack_require__.n(__WEBPACK_IMPORTED_MODULE_0_zone_js__);
+      /** PURE_IMPORTS_START zone_js PURE_IMPORTS_END */
+    `;
+    // tslint:enable:max-line-length
+
+    expect(oneLine`${purify(input)}`).toEqual(oneLine`${output}`);
+  });
+
+  it('prefix CCF and CMF statements', () => {
+    const input = stripIndent`
+      var AppComponentNgFactory = __WEBPACK_IMPORTED_MODULE_1__angular_core__["I" /* ɵccf */]('app-root');
+      var AppComponentNgFactory = __WEBPACK_IMPORTED_MODULE_1__angular_core__["I" /* ɵcmf */]('app-root');
```

Comment:
Nit: The duplicate var names are a bit confusing when scanning. It might be better to rename the second var to be a module factory since it's testing `createModuleFactory`.